lambda_function/ci.tf

```diff
@@ -0,0 +1,112 @@
+data "aws_ssm_parameter" "github_token" {
+  count = var.github_token_ssm_param == "" ? 0 : 1
+
+  name = var.github_token_ssm_param
+}
+
+resource "aws_codebuild_source_credential" "github_token" {
```

Comment:
The codebuild credential needs to be managed outside of the module because the AWS API seems to only allows creation of a single resource per auth, server type (https://docs.aws.amazon.com/cli/latest/reference/codebuild/import-source-credentials.html).  Having it inside the module like this results in different instances of this module stepping on/deleting..etc the same resource.   TJ and I discovered this in testing and he is pulling it out on his dev branch.

Reply:
fixed in https://github.com/PatientPing/terraform-aws-lambda/pull/3

Reply:
@terrancej I merged your branch into this one and then made some further updates.  Given that we can only have one GitHub credential per account (a ludicrous limitation IMO, but it is what it is) I think we can just construct the ARN.  I left in your codebuild_credential_arn argument as an optional backstop in case there is ever some future reason to override (e.g. CodeBuild is updated to let us have more than one GitHub credential).